### PR TITLE
[MU4] Added files statuses

### PR DESCRIPTION
**Pull request by igorkorsukov**
Added files statuses

### CODE REVIEW

File: src/autobot/autobottypes.h [hunk 1]
```diff
@@ -26,16 +26,9 @@
...
-    Undefined = 0,
-    None,
-    Success,
-    Failed
-};
-
 struct File {
     io::path path;
-    FileStatus status = FileStatus::Undefined;
+    Ret ret;
```
Comment (RomanPudashkin) [inline src/autobot/autobottypes.h:31]: Could you explain why you need ret here?

File: src/autobot/internal/autobot.h [hunk 2]
```diff
@@ -36,25 +36,31 @@ class Autobot : public IAutobot, public async::Asyncable
     std::vector<ITestCasePtr> testCases() const override;
     ITestCasePtr testCase(const std::string& name) const override;
 
-    void runAll(const std::string& testCaseName) override;
-    void runFile(const std::string& testCaseName, int fileIndex) override;
+    void setCurrentTestCase(const std::string& name) override;
+    const ValCh<ITestCasePtr>& currentTestCase() const override;
+
+    void runAll() override;
```
Comment (RomanPudashkin) [inline src/autobot/internal/autobot.h:42]: Maybe is it better to rename this method to runAllFiles?

File: src/autobot/autobottypes.h (hunk 1 above)
Comment (igorkorsukov) [reply src/autobot/autobottypes.h:31]: this is the result of the file test (in the future, there may be code and text message)

File: src/autobot/internal/autobot.h (hunk 2 above)
Comment (igorkorsukov) [reply src/autobot/internal/autobot.h]: renamed

**Review by RomanPudashkin — APPROVED** [review]

File: src/autobot/autobottypes.h (hunk 1 above)
Comment (RomanPudashkin) [reply src/autobot/autobottypes.h:31]: I would prefer to rename this to "testCaseRet" then :) Or rename this struct to FileTestResult (or something similar on your choice :))

**Pull request merged**